frontend/src/static/js/v_authorship.js

```diff
@@ -88,10 +89,23 @@ window.vAuthorship = {
 
     getRepoProps(repo) {
       if (repo) {
-        const author = repo.users.filter((user) => user.name === this.info.author);
-        if (author.length > 0) {
-          this.info.name = author[0].displayName;
-          this.filesLinesObj = author[0].fileTypeContribution;
+        if (this.info.isMergeGroup) {
+          // sum of all users' file type contribution
+          repo.users.forEach((author) => {
+            Object.entries(author.fileTypeContribution).forEach(([type, cnt]) => {
```

Comment:
Just a suggestion: would it help readability if we abstract out the logic in the inner loop into a separate method?
Perhaps a method that updates the filesLinesObj given a fileTypeContribution object?

Reply:
Yep that would be much cleaner. Refactored